Auth. ExpressJS API:

nosslsvec8 opened the pull request:
- add new user registration

- add new user authorization (Bearer Token)

- add user verification by token

nosslsvec8 committed:
Auth. ExpressJS API:

- add new user registration

- add new user authorization (Bearer Token)

- add user verification by token

ddavidov commented on api/routes/auth.js:
```diff
@@ -0,0 +1,53 @@
+
+    notEmpty(!email.trim(), 'Enter email');
+    notEmpty(!password.trim(), 'Enter password');
+    notEmpty(!name.trim(), 'Enter name');
```
...

ddavidov replied:
в том смысле,  что будет не очень очевидно как-то.

ddavidov commented on api/routes/user.js:
```diff
@@ -1,17 +1,24 @@
 });
-router.delete('/user', (req, res) => {
-    res.send('Delete user');
+router.delete('/user', async (req, res) => {
```
а вот тут и checkAuth не помешало б как минимум навесить

ddavidov commented on api/services/passport.js:
```diff
@@ -0,0 +1,34 @@
+passport.use(new BearerStrategy(
+    async function(token, done) {
+        const user = await User.findByToken(token);
+        console.log('CURRENT USER:', user);
```
:) Не нужен console.log тут больше

nosslsvec8 committed:
Auth. JWT Token:

- replacing the transition from Bearer Token to JWT Token

- added checking the author of a post when deleting or editing a post
